packages/grafana-ui/src/components/uPlot/hooks.test.ts

```diff
@@ -1,9 +1,11 @@
 import { usePlotConfig } from './hooks';
-import { renderHook, act } from '@testing-library/react-hooks';
+import { renderHook } from '@testing-library/react-hooks';
+import { act } from '@testing-library/react';
+import { UPlotConfigBuilder } from './config/UPlotConfigBuilder';
 
 describe('usePlotConfig', () => {
   it('returns default plot config', async () => {
-    const { result } = renderHook(() => usePlotConfig(0, 0, 'browser'));
+    const { result } = renderHook(() => usePlotConfig(0, 0, 'browser', new UPlotConfigBuilder()));
```

Comment:
@Clarity-89 I'm struggling to get this test to work. Would love to get some help from you. As you can see here https://github.com/grafana/grafana/blob/45b43598a6d6b41af9e8f3974cfd4fcf1e066e57/packages/grafana-ui/src/components/uPlot/hooks.ts the plugins hook uses rAF to make sure that all plugins have registered. The `arePluginsReady` flag is being used by the main `usePlotConfig` hook to indicate that [we are ready to create the final config for the uPlot](https://github.com/grafana/grafana/blob/45b43598a6d6b41af9e8f3974cfd4fcf1e066e57/packages/grafana-ui/src/components/uPlot/hooks.ts#L123). But I cannot get the rAF to work correctly in the test. I've tried couple of approaches with mocking rAF included, but it seems that the hook is being unmounted too soon, meaning the `arePluginsRead` never has a chance to be set to true. 

Any good practices, advice ho to aproach rAF with react-testing-library?

Reply:
can we comment out this so we can merge this?

Reply:
I think the best approach would still be to mock RAF, I'll check why it might not be working as expected.

Reply:
@torkelo yup, I'll comment this out for now and try to nail it down independently

Reply:
@dprokop so the issue here is not as much with the tests as with the dependencies array in usePlotConfig's `useEffect`. Basically `configBuilder` and `plugins` need to be memoized otherwise they seem to trigger multiple `useEffect` calls and the tests hang. For `configBuilder` it's quite easy, you just need to instantiate it before passing to the hook:
```
const builder = new UPlotConfigBuilder();
const { result } = renderHook(() => usePlotConfig(0, 0, 'browser', builder));
```
It's a bit more tricky with `plugins`, since it's a return value from another hook, haven't looked into that but quite sure it can be fixed. 
For mocking RAF I used this package: https://github.com/alexreardon/raf-stub. Also the hooks testing library has `waitForNextUpdate`, which has to be called in order for state updates to be properly applied. So the test would look like this:
```
import { usePlotConfig } from './hooks';
import { renderHook } from '@testing-library/react-hooks';
import { act } from '@testing-library/react';
import { UPlotConfigBuilder } from './config/UPlotConfigBuilder';

import { replaceRaf } from 'raf-stub';

replaceRaf();

describe('usePlotConfig', () => {
  it('returns default plot config', async () => {
    const builder = new UPlotConfigBuilder();
    const { result, waitForNextUpdate } = renderHook(() => usePlotConfig(0, 0, 'browser', builder));
    requestAnimationFrame.step();
    await waitForNextUpdate();

    expect(result.current.currentConfig).toMatchInlineSnapshot(`
      Object {
        "axes": Array [],
        "cursor": Object {
          "focus": Object {
            "prox": 30,
          },
        },
        "focus": Object {
          "alpha": 1,
        },
        "gutters": Object {
          "x": 8,
          "y": 8,
        },
        "height": 0,
        "hooks": Object {},
        "legend": Object {
          "show": false,
        },
        "plugins": Array [],
        "scales": Object {},
        "series": Array [
          Object {},
        ],
        "tzDate": [Function],
        "width": 0,
      }
    `);
  });
});
```

With this test and removing `plugins` from the dependency array, the test passes. Didn't look into the rest of the tests as they depend on plugins, but I'm sure you can take it from here :) Just need to make sure that `plugins` reference is stable during renders.

Reply:
@Clarity-89 thanks a lot for taking a look. For now I'm gonna merge this with test commented out and fix those in a followup PR